src/main/resources/avro/variants.avdl

```diff
@@ -178,42 +232,181 @@ record Variant {
 
   /**
-  The reference on which this variant occurs.
-  (e.g. `chr20` or `X`)
+  The `Allele`s available for this `Variant`. The first `Allele` is the
+  reference `Allele`, and the rest are alternate `Allele`s.
+  */
+  array<string> alleleIds;
+
+  /**
+  A map of additional variant information.
+  */
+  map<array<string>> info = {};
+}
+
+/**
+A `Call` represents the determination of genotype with respect to a
+particular `Allele`. It may include associated information such as quality
+and phasing. For example, a call might assign a probability of 0.32 to
+the occurrence of a SNP named rs1234 in a call set with the name NA12345.
+*/
+record Call {
+
+  /**
+  The ID of the call set this `Call` belongs to.
   */
-  string referenceName;
+  union { null, string } callSetId;
 
   /**
-  The start position at which this variant occurs (0-based).
-  This corresponds to the first base of the string of reference bases.
-  Genomic positions are non-negative integers less than reference length.
-  Variants spanning the join of circular genomes are represented as
-  two variants one on each side of the join (position 0).
+  The name of the call set this `Call` belongs to.
   */
-  long start;
+  union { null, string } callSetName = null;
 
   /**
-  The end position (exclusive), resulting in [start, end) closed-open interval.
-  This is typically calculated by `start + referenceBases.length`.
+  The ID of the `Allele` that this `Call` reflects the presence of.
   */
-  long end;
+  string alleleId;
 
-  /** The reference bases for this variant. They start at the given position. */
-  string referenceBases;
+  /**
+  The ID of the `Variant` in the context of which this `Call` is calling the
+  copy number on the `Allele`. The `Allele` called by this `Call` must be on the
+  list of `Allele` IDs that are allowable alternatives for the `Variant`. Only
+  one `Call` in a `CallSet` can describe a given `Allele` within a particular
+  `Variant`.
+  */
+  union { null, string } variantId = null;
+
+  // HAPLOTYPE-type CallSets double as phase sets, so we don't need phase sets.
 
-  /** The bases that appear instead of the reference bases. */
-  array<string> alternateBases = [];
+  // We have various ways of specifying what the call actually is.
+  // Not all of the following will be present, but at least one should be
 
   /**
-  A map of additional variant information.
+  0 for not present in callSet, 1 for single copy, 2 for 2 copies etc.
+  */
+  union { null, int } copyNumber = null;
+
+  /**
+  Expected value of copyNumber.
+  */
+  union { null, float } dosage = null;
+
+  /**
+  Probability that copy number is k: 0..n
+  */
+  array<float> probability = [];
+
+  /**
+  Likelihood of data given copy number k.
+  */
+  array<float> likelihood = [];
+
+  /**
+  Number of reads supporting this allele (DP in VCF).
+  */
+  union { null, int } readCount = null;
+
+  /**
+  Potential other evidence for the call.
+  */
+  map<string> evidence = {};
```

Comment:
I don't think `evidence` actually does anything useful that can't be rolled into the `info` map. We could separate certain keys out into `evidence` in our spec, but at that point we would probably just want to standardize them as fields on `Call`.

I'll just cut `evidence`, and any other information you need to bring along, whether it is evidence for the call or not, will go in `info`.